src/EntityImporter.php

```diff
@@ -9,181 +9,238 @@
+        StatementsCountLookup $statementsCountLookup,
+        LoggerInterface $logger
+    )
+    {
+        $this->statementsImporter = $statementsImporter;
+        $this->badgeItemUpdater = $badgeItemUpdater;
+        $this->apiEntityLookup = $apiEntityLookup;
+        $this->entityStore = $entityStore;
+        $this->entityMappingStore = $entityMappingStore;
+        $this->statementsCountLookup = $statementsCountLookup;
+        $this->logger = $logger;
+
+        $this->idParser = new BasicEntityIdParser();
+        $this->importUser = User::newFromId(0);
+        $this->batchSize = 10;
+    }
+
+    public function importEntities(array $ids, $importStatements = true)
+    {
+        $batches = array_chunk($ids, $this->batchSize);
+
+        $stashedEntities = array();
+
+        foreach ($batches as $batch) {
+            $entities = $this->apiEntityLookup->getEntities($batch);
+
+            if ($entities) {
+                $this->importBadgeItems($entities);
+            } else {
+                $this->logger->error('Failed to retrieve items for batch');
+            }
+
+            $stashedEntities = array_merge($stashedEntities, $this->importBatch($batch));
+        }
+
+        if ($importStatements === true) {
+            foreach ($stashedEntities as $entity) {
+                $referencedEntities = $this->getReferencedEntities($entity);
+                $this->importEntities($referencedEntities, false);
+
+                $entity_new = $entity;
+                $statements_new = $entity->getStatements();
+                $uri = WikibaseRepo::getDefaultInstance()->getSettings()->getSetting('conceptBaseUri');
+                foreach ($statements_new as $key1 => $statement_new) {
+                    $snak_new = $statement_new->getMainSnak();
+                    if ($snak_new instanceof PropertyValueSnak) {
+                        $data_value_new = $snak_new->getDataValue();
+                        if ($data_value_new instanceof UnboundedQuantityValue) {
+                            $unit = $data_value_new->getUnit();
+                            if (strpos($unit, 'http://www.wikidata.org/entity/') !== false) {
+                                $id = str_replace("http://www.wikidata.org/entity/", "", $unit);
+                                $newid = $this->entityMappingStore->getLocalId(new ItemId($id));
+                                $data_value_new = new UnboundedQuantityValue($data_value_new->getAmount(), $uri . $newid);
+                                $snak_new = new PropertyValueSnak($snak_new->getPropertyId(), $data_value_new);
+                                $statement_new->setMainSnak($snak_new);
+                            }
+                        }
+                    }
+                    $snakList_new = $statement_new->getQualifiers();
+                    foreach ($snakList_new as $key2 => $snak_new) {
+                        if ($snak_new instanceof PropertyValueSnak) {
```

Comment:
You can shorten this our by flipping the condition and doing a continue inside or extract this to a method. This is too much indentation

Reply:
what you mean by flipping the condition?

Reply:
I mean something like this:
```lang=php
if (!$snak_new instanceof PropertyValueSnak) {
confinute;
}
```